### PR TITLE
feat(auditor): restart webservice on client request

### PR DESCRIPTION
**Pull request by maqi**
### Description

auditor restart webservice on client request

### CODE REVIEW

File: sn_auditor/src/main.rs [hunk 1]
```diff
@@ -262,43 +262,58 @@ async fn initialize_background_spend_dag_collection(
...
+            // Dispatch the request to the appropriate handler
+            let response = match request.url() {
+                "/" => routes::spend_dag_svg(&dag),
+                s if s.starts_with("/spend/") => routes::spend(&dag, &request).await,
+                s if s.starts_with("/add-participant/") => {
+                    routes::add_participant(&dag, &request).await
+                }
+                "/beta-rewards" => routes::beta_rewards(&dag).await,
+                "/restart" => {
+                    info!("Restart auditor web serive as to client's request");
```
Comment (mazzi) [inline sn_auditor/src/main.rs]: typo: `serive` vs `service`

File: sn_auditor/src/main.rs [hunk 2]
```diff
@@ -262,43 +262,58 @@ async fn initialize_background_spend_dag_collection(
...
+                    break;
+                }
+                "/terminate" => {
+                    info!("Terminate auditor web serive as to client's request");
```
Comment (mazzi) [inline sn_auditor/src/main.rs]: same

**Comment by mazzi** [comment]
I don't get how `break` and `return Ok()` can effectively restart or terminate the service =\

**Comment by maqi** [comment]
> I don't get how `break` and `return Ok()` can effectively restart or terminate the service =\

the `server` instance got destructed and re-constructed

**Review by mazzi — APPROVED** [review]

**Comment by maqi** [comment]
replaced by https://github.com/maidsafe/safe_network/pull/2206
as in favour of merge into main

**Comment by maqi** [comment]
replaced by https://github.com/maidsafe/safe_network/pull/2207
as in favour of merge into stable branch

**Pull request closed without merging**